app/controllers/foreman_wreckingball/hosts_controller.rb

```diff
@@ -52,22 +52,21 @@ def status_dashboard
 
     # ajax method
     def status_hosts
-      statuses_map = {
-        vmware_tools_status_object: ForemanWreckingball::ToolsStatus,
-        vmware_operatingsystem_status_object: ForemanWreckingball::OperatingsystemStatus,
-        vmware_cpu_hot_add_status_object: ForemanWreckingball::CpuHotAddStatus,
-        vmware_spectre_v2_status_object: ForemanWreckingball::SpectreV2Status,
-        vmware_hardware_version_status_object: ForemanWreckingball::HardwareVersionStatus
-      }
-
-      @status = statuses_map[params[:status].to_sym]
+      @status = STATUSES_MAP[params.fetch(:status).to_sym]
+
+      total = Host.authorized(:view_hosts, Host)
+                  .joins(@status.host_association)
+                  .includes(@status.host_association, :vmware_facet, :environment)
+                  .count
+
       all_hosts = Host.authorized(:view_hosts, Host)
                       .joins(@status.host_association)
                       .includes(@status.host_association, :vmware_facet, :environment)
+                      .where('"host_status"."status" NOT IN (:status)', status: @status.global_ok_list)
```

Comment:
Can this be written like this?

```ruby
.where.not('host_status.status': @status.global_ok_list)
```

Reply:
Yes, good idea, I tried the `.where.not` version before, somehow it was not generating a valid SQL query.